src/_P037_MQTTImport.ino

```diff
@@ -53,6 +59,75 @@ String P037_getMQTTLastTopicPart(const String& topic) {
 
 # endif // if P037_MAPPING_SUPPORT || P037_JSON_SUPPORT
 
+bool P037_addEventToQueue(struct EventStruct *event, String& newEvent) {
+  if (newEvent.isEmpty()) { return false; }
+  bool result    = true;
+  uint8_t reason = 0;
+
+  if (result && P037_DEDUPLICATE_EVENTS) { // Check if event is already queued
+    String queuedEvent;
+    reason = 1;                            // Deduping started, so that might be the reason
+    # if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+    String log;
+
+    if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+      addLog(LOG_LEVEL_DEBUG, F("P037: addEvent deduping"));
+      addLog(LOG_LEVEL_DEBUG, newEvent);
+      log.reserve(64);
+    }
+    # endif // if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+    for (auto iter = eventQueue.begin(); iter != eventQueue.end() && result; ++iter) {
+      queuedEvent = String(*iter);
+      # if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+      if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+        log += queuedEvent;
+        log += ';';
+      }
+      # endif // if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+      result = !queuedEvent.equalsIgnoreCase(newEvent); // IgnoreCase-compare!
+
+      # if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+
+      if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+        log += result ? '+' : '-';
+        addLog(LOG_LEVEL_DEBUG, log);
+        log.clear();
+      }
+      # endif // if !defined(LIMIT_BUILD_SIZE) || defined(P037_OVERRIDE)
+    }
+  }
+
+  if (result) {
+    if ((P037_QUEUEDEPTH_EVENTS == 0) || (eventQueue.size() <= P037_QUEUEDEPTH_EVENTS)) {
+      eventQueue.add(newEvent);
+    } else {
+      result = false;
+      reason = 2; // Queue limit reached
+    }
+  }
+
+  if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+    String log = F("MQTT: Event added: ");
```

Comment:
Please wrap these logs in `#ifndef BUILD_NO_DEBUG`

Reply:
I wanted this one to always be available, that's why I left that #define out.